src/Nancy/HttpLink.cs

```diff
@@ -0,0 +1,283 @@
+        /// <exception cref="System.ArgumentNullException"><paramref name="targetUri" /> or <paramref name="relation" /></exception>
+        public HttpLink(string targetUri, string relation, string type)
+            : this(ParseUri(targetUri), relation, type, string.Empty)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HttpLink" /> class.
+        /// </summary>
+        /// <param name="targetUri">The target URI of the link.</param>
+        /// <param name="relation">The relation that identifies the semantics of the link.</param>
+        /// <param name="type">The optional type parameter is a hint indicating what the media type of the result of dereferencing the link should be. Note that this is only a hint; for example, it does not override the HTTP Content-Type header of a HTTP response obtained by actually following the link.</param>
+        /// <param name="title">The optional title parameter is used to label the destination of a link such that it can be used as a human-readable identifier (e.g., a menu entry) in the language indicated by the HTTP Content-Language header (if present).</param>
+        /// <exception cref="System.ArgumentNullException"><paramref name="targetUri" /> or <paramref name="relation" /></exception>
+        public HttpLink(string targetUri, string relation, string type, string title)
+            : this(ParseUri(targetUri), relation, type, title)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HttpLink" /> class.
+        /// </summary>
+        /// <param name="targetUri">The target URI of the link.</param>
+        /// <param name="relation">The relation that identifies the semantics of the link.</param>
+        /// <param name="type">The optional type parameter is a hint indicating what the media type of the result of dereferencing the link should be. Note that this is only a hint; for example, it does not override the HTTP Content-Type header of a HTTP response obtained by actually following the link.</param>
+        /// <param name="title">The optional title parameter is used to label the destination of a link such that it can be used as a human-readable identifier (e.g., a menu entry) in the language indicated by the HTTP Content-Language header (if present).</param>
+        /// <exception cref="System.ArgumentNullException"><paramref name="targetUri" /> or <paramref name="relation" /></exception>
+        public HttpLink(Uri targetUri, string relation, string type, string title)
+        {
+            if (targetUri == null)
+            {
+                throw new ArgumentNullException("targetUri");
+            }
+
+            if (relation == null)
+            {
+                throw new ArgumentNullException("relation");
+            }
+
+            this.parameters = new Dictionary<string, object>
```

Comment:
Good question. mnot/I-D@825bf92424a977a040f69f2945b9f1d4a24c9504 attempts to clarify this a bit, by basically saying nothing. RFC 5988 itself defines all attributes as case insensitive and singular, though (see mnot/I-D#170 for clearer language on this), and I believe that to be the case with most other applications of the `Link` header as well. 

So, the most "correct" implementation would probably be something like `IEnumerable<KeyValuePair<string, string>>`, but the more pragmatic and usable thing would be to initialize a `Dictionary<string, string>` with `StringComparer.InvariantCultureIgnoreCase`.